### PR TITLE
Vyos route maps

### PR DESCRIPTION
**Pull request by ashwini-mhatre**
##### ISSUE TYPE

- New Module Pull Request

##### COMPONENT NAME

vyos_route_maps

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Review by GomathiselviS — CHANGES_REQUESTED** [review]

File: tests/integration/test-inventory [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+ansible_ssh_pass=vyos
+ansible_connection=ansible.netcommon.network_cli
+ansible_become=true
+#ansible_ssh_private_key_file=/Users/amhatre/.ssh/id_rsa
+ansible_become_method=ansible.netcommon.enable	
+cli={'transport': 'cli'}
+netconf={'transport': 'netconf'}
+
+[netconf]
+iosxr-7.0.2  ansible_port=22 ansible_host=127.0.0.1 ansible_user=vagrant
```
[the remark this answers is not in the document]
Comment (ashwini-mhatre) [reply tests/integration/test-inventory:17]: added by mistake

File: plugins/modules/vyos_route_maps.py [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+      - deleted
+      - merged
+      - overridden
+      - replaced
+      - gathered
+      - rendered
+      - parsed
+      default: merged
+"""
+EXAMPLES = """
```
[the remark this answers is not in the document]
Comment (ashwini-mhatre) [reply plugins/modules/vyos_route_maps.py:261]: ack

File: plugins/module_utils/network/vyos/config/route_maps/route_maps.py [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+
+    def _compare(self, want, have):
+        """Leverages the base class `compare()` method and
+           populates the list of commands to be run by comparing
+           the `want` and `have` data with the `parsers` defined
+           for the Route_maps network resource.
+        """
+        w_entries = want.get("entries", {})
+        h_entries = have.get("entries", {})
+        self._compare_entries(want=w_entries, have=h_entries)
```
[the remark this answers is not in the document]
Comment (ashwini-mhatre) [reply plugins/module_utils/network/vyos/config/route_maps/route_maps.py:144]: outside entries we only have route_map key . and we only remove route_map when  performing delete and overridden operation which is taken cared through 125 to 135 code

File: plugins/module_utils/network/vyos/config/route_maps/route_maps.py [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+
+    def _compare_entries(self, want, have):
+        for wk, wentry in iteritems(want):
+            hentry = have.pop(wk, {})
+            self.compare(parsers=self.parsers, want=wentry, have=hentry)
+
+        # remove superfluos entries from have
+        #for _hk, hentry in iteritems(have):
+        #   self.commands.append(self._tmplt.render(hentry, "route_map", True))
+
```
[the remark this answers is not in the document]
Comment (ashwini-mhatre) [reply plugins/module_utils/network/vyos/config/route_maps/route_maps.py:154]: this is taken care in compare method as vyos has linear structure we dont need to handle have entries. please check examples for more details

File: plugins/module_utils/network/vyos/rm_templates/route_maps.py [hunk 5]
```diff
@@ -0,0 +1,1293 @@
...
+            "name": "route_map",
+            "getval": re.compile(
+                r"""
+                ^set\spolicy\sroute-map\s(?P<route_map>\S+)
+                *$""",
+                re.VERBOSE,
+            ),
+            "compval": "route_map",
+            "setval": "policy route-map {{route_map}}",
+            "remval": "policy route-map {{route_map}}",
```
[the remark this answers is not in the document]
Comment (ashwini-mhatre) [reply plugins/module_utils/network/vyos/rm_templates/route_maps.py:42]: ack

File: plugins/module_utils/network/vyos/argspec/route_maps/route_maps.py [hunk 6]
```diff
@@ -0,0 +1,224 @@
...
+                "entries": {
+                    "aliases": ["rules"],
+                    "type": "list",
+                    "elements": "dict",
+                    "options": {
+                        "rule_number": {"type": "int"},
+                        "call": {"type": "str"},
+                        "description": {"type": "str"},
+                        "action": {"type": "str", "choices": ["deny", "permit"]},
+                        "continue": {"type": "int"},
```
[the remark this answers is not in the document]
Comment (ashwini-mhatre) [reply plugins/module_utils/network/vyos/argspec/route_maps/route_maps.py:54]: ack

File: plugins/module_utils/network/vyos/argspec/route_maps/route_maps.py [hunk 7]
```diff
@@ -0,0 +1,224 @@
...
+            "type": "list",
+            "elements": "dict",
+            "options": {
+                "route_map": {"type": "str"},
+                "entries": {
+                    "aliases": ["rules"],
+                    "type": "list",
+                    "elements": "dict",
+                    "options": {
+                        "rule_number": {"type": "int"},
```
[the remark this answers is not in the document]
Comment (ashwini-mhatre) [reply plugins/module_utils/network/vyos/argspec/route_maps/route_maps.py]: ack

File: tests/integration/network-integration.cfg [hunk 8]
```diff
@@ -2,3 +2,11 @@
...
 connect_timeout = 100
 connect_retry_timeout = 100
+
+
+[paramiko_connection]
+look_for_keys = False
+
+[defaults]
+gathering=no
+interpreter_python=/Users/amhatre/ansible_venvs/py3.6.10/bin/python
```
[the remark this answers is not in the document]
Comment (ashwini-mhatre) [reply tests/integration/network-integration.cfg]: removed

File: plugins/modules/vyos_route_maps.py [hunk 9]
```diff
@@ -0,0 +1,953 @@
...
+The module file for vyos_route_maps
+"""
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+DOCUMENTATION = """
+module: vyos_route_maps
+version_added: 2.0.2
```
Comment (NilashishC) [inline plugins/modules/vyos_route_maps.py:17]: This should be the next minor version.

**Comment by ashwini-mhatre** [comment]
recheck

**Review by GomathiselviS — APPROVED** [review]

**Pull request merged**